### PR TITLE
Feature/145 external housing available base data structure

### PR DESCRIPTION
**Pull request by JaniceHe26**
Added External Housing Available Base Data Structure.

### CODE REVIEW

**Comment by zaclittleberry** [comment]
# All
status should just be a Sequelize.STRING. enum or something else isn't necessarily a bad choice, but we don't know what those values will be and string will be more flexible for us.

the module you added should be added to lib/modules.js. You might get compilation errors after that because now your code is actually being run and it has problems. This is the right path to take, and work through those errors.

*.routes.js and *.controllers.js should have the outer structure of those files in them. basically, the module.exports statement, and the bare structure to get to their return statments. the returns can just be [] and {} for routes and controllers respectively.

AuthorId and location will be added via association. they don't get defined as fields on the 'main' part of the model.

# Janice

I know it's verbose, but can you change ExternalHousing to ExternalHousingAvailable. Otherwise it will end up conflicting with ExternalHousingWanted if/when we make that (if we followed the same pattern).

looks like there's no newline at the end of your files. This is preferred by git. There might be a setting in your editor to do this automatically when you save files.

numberOfBeds and numberOfBaths can be changed to just beds and baths

no need for the id and api models right now, should be removed. We'll add those when we get to routes.

**Review by zaclittleberry — CHANGES_REQUESTED** [review]
see comments in conversation tab

**Review by VitaRox — COMMENTED** [review]
Looks good to me!  It makes me wonder if I should've included those imports in my skeletal controller and routes.js files.  Anyway, looks great!

File: lib/external-housing-available/external-housing-available.models.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+module.exports = {
+  db: (sequelize, Sequelize) => {
+    const ExternalHousingAvailable = sequelize.define('ExternalHousingAvailable', {
+      id: {
+        type: Sequelize.UUID,
+        defaultValue: Sequelize.UUIDV4,
+        primaryKey: true
+      },
+      author: Sequelize.STRING,
```
Comment (zaclittleberry) [inline lib/external-housing-available/external-housing-available.models.js]: please remove. should be an association, not defined this way.

File: lib/external-housing-available/external-housing-available.models.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      title: Sequelize.STRING,
+      address: Sequelize.STRING,
+      location: Sequelize.STRING,
```
Comment (zaclittleberry) [inline lib/external-housing-available/external-housing-available.models.js]: please remove. should be an association, not defined this way.

File: lib/external-housing-available/external-housing-available.models.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      contact: Sequelize.STRING,
+      body: Sequelize.TEXT,
+      status: Sequelize.STRING,
+      url: Sequelize.TEXT,
+      details: {
+        type: Sequelize.JSONB,
+        defaultValue: {},
+      },
+    });
+    return ExternalHousingAvailable;
```
Comment (zaclittleberry) [inline lib/external-housing-available/external-housing-available.models.js:31]: we'll still need to define associations for this model. We can go over this as a group. I can't remember if I told you all to hold off on that or give it a shot. my apologies.

**Review by zaclittleberry — CHANGES_REQUESTED** [review]
looks very clean. just a few lines to remove from the model, and a few things to add as associations. you can try this before mtg if you have time and then we can go over it as a group. good job. (author and location should have examples on the property content type you can pull from).

**Review by zaclittleberry — APPROVED** [review]

**Pull request merged**